nameserver/mdns_client.go

```diff
@@ -93,7 +117,13 @@ func (c *MDNSClient) Start(ifi *net.Interface) error {
 	actionChan := make(chan MDNSAction, MailboxSize)
 	c.actionChan = actionChan
 	go c.actorLoop(actionChan)
+	return nil
+}
 
+func (c *MDNSClient) Stop() error {
```

Comment:
We didn't have a proper shutdown mechanism. And it could be improved by waiting for the actor goroutine...